asapdiscovery-alchemy/asapdiscovery/alchemy/utils.py

```diff
@@ -81,14 +82,46 @@ def create_network(
         result_network = FreeEnergyCalculationNetwork(**network_data)
         return result_network
 
+    def get_actioned_weights(self) -> list[float]:
+        """
+        For all waiting/running networks, return the weights associated with them.
+
+        Args:
+
+        Returns:
+            A list of floats that are associated with all currently waiting/running networks.
+        """
+        finished_counter = 0
+        active_network_weights = []
+
+        # get all networks that we're able to see
+        for key in self._client.query_networks():
+            n_running = self._client.get_network_status(
+                network=key, visualize=False
+            ).get("running", 0)
+            n_waiting = self._client.get_network_status(
+                network=key, visualize=False
+            ).get("waiting", 0)
+
+            # if we've encountered 5 stale networks we've probably reached the finished ones, can safely stop collecting weights
+            if n_running == 0 and n_waiting == 0:
+                finished_counter += 1
+            if finished_counter == 5:
+                break
+
+            active_network_weights.append(self._client.get_network_weight(network=key))
+
+        return active_network_weights
+
     def action_network(
-        self, planned_network: FreeEnergyCalculationNetwork
+        self, planned_network: FreeEnergyCalculationNetwork, prioritize: Optional[bool]
```

Comment:
```suggestion
        self, planned_network: FreeEnergyCalculationNetwork, prioritize: Optional[bool]=None
```
with current setup need to specify None if we want default case to be  weight=0.5 (default). See other comment as well though.